zebra-network/src/peer/handshake.rs

```diff
@@ -1312,8 +1313,10 @@ async fn send_periodic_heartbeats_run_loop(
     while let Some(_instant) = interval_stream.next().await {
         // We've reached another heartbeat interval without
         // shutting down, so do a heartbeat request.
+        let ping_sent_at = Instant::now();
+
         let heartbeat = send_one_heartbeat(&mut server_tx);
-        heartbeat_timeout(heartbeat, &heartbeat_ts_collector, &connected_addr).await?;
+        let rtt = heartbeat_timeout(heartbeat, &heartbeat_ts_collector, &connected_addr).await?;
```

Comment:
There is an issue here: this only returns _after_ the pong was received, so the `ping_sent_at` is only written to the address book at that time, but then it's too late (its whole purpose is to keep track how much time has elapsed since the last ping was sent).

I'm not 100% sure how to fix this, but it might envolve sending a separate `MetaAddrChange` to the `heartbeat_ts_collector` before sending the ping here. I think we'll need a new `MetaAddrChange` type because I don't think the current ones would fit.

Aditionally, after the pong is received, we will need to actually clear the `ping_sent_at` value, so the `send()` below should send `None`.